src/sql/src/plan/statement/ddl/connection.rs

```diff
@@ -0,0 +1,490 @@
+                    Some(url) => url
+                        .parse()
+                        .map_err(|e| sql_err!("parsing schema registry url: {e}"))?,
+                    None => sql_bail!("invalid CONNECTION: must specify URL"),
+                };
+                let _ = url
+                    .host_str()
+                    .ok_or_else(|| sql_err!("invalid CONNECTION: URL must specify domain name"))?;
+                if url.path() != "/" {
+                    sql_bail!("invalid CONNECTION: URL must have an empty path");
+                }
+                let cert = self.ssl_certificate;
+                let key = self.ssl_key.map(|secret| secret.into());
+                let tls_identity = match (cert, key) {
+                    (None, None) => None,
+                    (Some(cert), Some(key)) => Some(TlsIdentity { cert, key }),
+                    _ => sql_bail!(
+                        "invalid CONNECTION: reading from SSL-auth Confluent Schema Registry requires both SSL KEY and SSL CERTIFICATE"
+                    ),
+                };
+                let http_auth = self.user.map(|username| CsrConnectionHttpAuth {
+                    username,
+                    password: self.password.map(|secret| secret.into()),
+                });
+
+                let tunnel = scx.build_tunnel_definition(self.ssh_tunnel, self.aws_privatelink)?;
+
+                Connection::Csr(CsrConnection {
+                    url,
+                    tls_root_cert: self.ssl_certificate_authority,
+                    tls_identity,
+                    http_auth,
+                    tunnel,
+                })
+            }
+            CreateConnectionType::Postgres => {
+                let cert = self.ssl_certificate;
+                let key = self.ssl_key.map(|secret| secret.into());
+                let tls_identity = match (cert, key) {
+                    (None, None) => None,
+                    (Some(cert), Some(key)) => Some(TlsIdentity { cert, key }),
+                    _ => sql_bail!(
+                        "invalid CONNECTION: both SSL KEY and SSL CERTIFICATE are required"
+                    ),
+                };
+                let tls_mode = match self.ssl_mode.as_ref().map(|m| m.as_str()) {
+                    None | Some("disable") => tokio_postgres::config::SslMode::Disable,
+                    // "prefer" intentionally omitted because it has dubious security
+                    // properties.
+                    Some("require") => tokio_postgres::config::SslMode::Require,
+                    Some("verify_ca") | Some("verify-ca") => {
+                        tokio_postgres::config::SslMode::VerifyCa
+                    }
+                    Some("verify_full") | Some("verify-full") => {
+                        tokio_postgres::config::SslMode::VerifyFull
+                    }
+                    Some(m) => sql_bail!("invalid CONNECTION: unknown SSL MODE {}", m.quoted()),
+                };
```

Comment:
Untested according to coverage data

Reply:
This is just code movement, so not adding tests for this.